### PR TITLE
Safe navigation in chat inbox

### PR DESCRIPTION
**Pull request by buoyad**
Fixes the issue where you quick-tap two rows on mobile and get two convos pushed. cc @keybase/y2ksquad

### CODE REVIEW

**Comment by buoyad** [comment]
I'm moving where the `navKey` comes from out of `MakeRow` -> `Inbox`

**Comment by buoyad** [comment]
k @chrisnojima `makeRow` is a function again, inbox is grabbing the key for all the rows.

File: shared/router-v2/util.tsx [hunk 1]
```diff
@@ -13,13 +14,15 @@ export const getActiveIndex = (navState: NavState): number => {
   return navState.index
 }
 // Get active key inside any subnavigator so navigation closures are
-export const getActiveKey = (navState: NavState): string => {
-  if (!navState.routes) {
-    return navState.key
+export const getActiveKey = memoize(
```
Comment (chrisnojima) [inline shared/router-v2/util.tsx]: does this memoizing work at all. this thing is called recursively so you're blowing it away right

Comment (buoyad) [reply shared/router-v2/util.tsx]: Good catch, fixed.

File: shared/router-v2/util.tsx [hunk 2]
```diff
@@ -1,4 +1,5 @@
 import {NavState} from '../constants/types/route-tree'
+import {memoize} from '../util/memoize'
```
Comment (chrisnojima) [inline shared/router-v2/util.tsx]: 🔥

**Pull request merged**